src/VisualStudio/Core/Impl/ProjectSystem/CPS/CPSProject_IWorkspaceProjectContext.cs

```diff
@@ -39,7 +38,7 @@ private void ExecuteForegroundAction(Action action)
                     _foregroundTaskQueue = _foregroundTaskQueue.SafeContinueWithFromAsync(
                         async _ =>
                         {
-                            await ThreadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(alwaysYield: true);
+                            await ThreadingContext.JoinableTaskFactory.SwitchToMainThreadAsync();
```

Comment:
it's terrifying to me that this is all being done under a lock and SafeContinueWithFromAsync is called with ExecuteSynchronously.  My understanding is that means that it can start executing its lambda immediately if the preceding task is completed.  

Even if this works, i hate all this code.

Reply:
Note: it would probably mike me feel *slightly* better if this had an explicit RunContinuationsAsynchronously.  But i'm not going to press the issue :)

Reply:
> Note: it would probably mike me feel slightly better if this had an explicit RunContinuationsAsynchronously.

That's what the code currently does. This pull request removes that behavior as an optimization.

Reply:
I mean specifically in the TaskContinuationOptions.  I get that it should be effectively the same.  I just hate seeing code potentially run inside a lock like this.

Reply:
(Note: don't actually make any changes here).  I'm just lamenting out loud.  I do hopw that all your future work will end up making this quite clear and safe.